faasmcli/faasmcli/util/policy.py

```diff
@@ -0,0 +1,218 @@
+                    if verify_objects[next_id] == 'end: {}:{}'.format(name, id):
+                        _found = True
+                        break
+                if not _found:
+                    return False
+                current_index += 1
+            elif meta_char == '<':
+                _count = 0
+                for _index, _char in enumerate(tmp_stack):
+                    if _char == '<':
+                        _count += 1
+                    if _char == '>':
+                        if _count == 0:
+                            break
+                        else:
+                            _count -= 1
+                current_index += _index
+                new_stack = tmp_stack[0:_index]
+                split = new_stack.split(',')
+                if len(split) == 1:
+                    func = split[0][1:-1]
+                else:
+                    func = split[0][1:]
+                _found = False
+                _split = func.split(':')
+                func = "{}:{}".format(_split[0], _split[2])
+                for next_id in next_function:
+                    if verify_objects[next_id] == func:
+                        next_function = verify_edges[next_id]
+                        _found = True
+                        break
+                if not _found:
+                    return False
+                func_ = _split[0]
+                _id = _split[2]
+                if not self._is_valid_execution(new_stack, func_, _id):
+                    return False
+            elif meta_char == "[": #state
+                _count = 0
+                for _index, _char in enumerate(tmp_stack):
+                    if _char == '[':
+                        _count += 1
+                    if _char == ']':
+                        if _count == 0:
+                            break
+                        else:
+                            _count -= 1
+                current_index += _index
+                state = tmp_stack[0:_index][1:-1]
+                state_key, execution_stack = state.split(';') #cleanup
+                reversed_execution_stack = execution_stack[::-1]
+                _count = 0
+                for _index, _char in enumerate(reversed_execution_stack):
+                    if _char == '>':
+                        _count += 1
+                    if _char == '<':
+                        _count -= 1
+                        if _count < 0:
+                            break
+                split = reversed_execution_stack[0:_index][::-1].split(':')
+                last_writer_func = split[0]
+                writer_hash = split[1]
+                last_writer_id = split[2][0] #TODO id > 10...
+                test = list(reversed(self._verify_chain[last_writer_func][str(int(last_writer_id))]['nodes']))
+                for _index, _label in enumerate(list(reversed(self._verify_chain[last_writer_func][str(int(last_writer_id))]['nodes']))):
```

Comment:
This file is full of one-liners like this that are incredibly difficult to understand. Some will get picked up by `black` and `flake8`, but could you also just split up stuff like this aiming for one operation per line?